commons-hpcc/src/main/java/org/hpccsystems/commons/ecl/RecordDefinitionTranslator.java

```diff
@@ -758,6 +785,17 @@ private static int getJsonTypeDefinition(FieldDef field, HashMap<Integer, Intege
             }
         }
 
+        if (field.isNonStandardInt())
+        {
+            FieldDef nonKeyedField = new FieldDef(field);
+            nonKeyedField.setSourceType(HpccSrcType.LITTLE_ENDIAN);
+
+            int childTypeHash = getJsonTypeDefinition(nonKeyedField, typeDefinitionMap, typeDefinitions);
+            int childTypeIndex = typeDefinitionMap.get(childTypeHash);
+            String childTypeName = "ty" + (childTypeIndex + 1);
```

Comment:
it's difficult to rationalize the significance of this literal "ty" let's externalize into a self documenting var